apps/emqx_connector/src/mqtt/emqx_connector_mqtt_worker.erl

```diff
@@ -321,17 +325,15 @@ common(_StateName, {call, From}, get_forwards, #{connect_opts := #{forwards := F
     {keep_state_and_data, [{reply, From, Forwards}]};
 common(_StateName, {call, From}, get_subscriptions, #{connection := Connection}) ->
     {keep_state_and_data, [{reply, From, maps:get(subscriptions, Connection, #{})}]};
-common(_StateName, info, {deliver, _, Msg},
-       State = #{replayq := Q, if_record_metrics := IfRecordMetric}) ->
+common(_StateName, info, {deliver, _, Msg}, State = #{replayq := Q}) ->
```

Comment:
why is this clause still necessary?

Reply:
If the MQTT Bridge is started without rule engine, the mqtt_worker still subscribes to the local broker, so it will receive messages delivered from the local broker.